program-error/Cargo.toml

```diff
@@ -11,19 +11,24 @@ edition = { workspace = true }
 
 [dependencies]
 borsh = { workspace = true, optional = true }
-num-traits = { workspace = true }
+num-traits = { workspace = true, optional = true }
 serde = { workspace = true, optional = true }
 serde_derive = { workspace = true, optional = true }
-solana-decode-error = { workspace = true }
-solana-instruction = { workspace = true, default-features = false, features = [
-    "std",
-] }
-solana-msg = { workspace = true }
-solana-pubkey = { workspace = true, default-features = false }
+solana-decode-error = { workspace = true, optional = true }
+solana-instruction-error = { workspace = true }
```

Comment:
Can we make this one optional?

Reply:
are you sure? I think this is at the point where the cognitive overhead of the extra feature outweighs any impact on compile times or API surface area

Reply:
Yeah, the amount of features to enable/disable depending of whether you are using pinocchio or the SDK concerns me - hopefully is not going to be annoying.

Ideally I would not have `solana-instruction-error` and `solana-pubkey-error` exported here – those should be added by the code that wants to use them. So the other option is to remove the export from here, instead of adding another feature.

Reply:
pinocchio would want the error codes defined in solana-instruction-error e.g. `pub const INVALID_ARGUMENT: u64 = to_builtin!(2);`. Currently it's redefining these itself. Are you saying you'd want these consts to be in a standalone crate without the InstructionError enum?

Reply:
I was thinking that we can have the error codes defined in `solana-program-error`, like in pinocchio. So `solana-instruction-error` is the one that depends on `solana-program-error`, not the other way around. The error codes cover the "program errors" variants and `InstructionError` is a super set of `ProgramError`.

Reply:
ok i've moved some stuff to program-error